src/components/MultiSelect.tsx

```diff
@@ -99,9 +95,16 @@ const MultiSelect = (props: MultiSelectProperties): React.JSX.Element => {
         }}
         data={updatePriorities(props.data)}
         type="multi"
-        onSelect={selectItem}
+        onSelect={props.onSelect}
+        onRemove={props.onRemove}
         selected={selected}
-        clearSelected={() => setSelected([])}
+        setSelected={setSelected}
+        clearSelected={() => {
+          for (const item of selected) {
+            props.onRemove?.(item);
+          }
```

Comment:
clearSelected currently loops all the selected items even if onRemove is not defined/passed.

```suggestion
          if (props.onRemove) {
            selected.forEach(props.onRemove);
          }
```

Reply:
I agree with this change, I'll commit this suggestion.